funnel/views/index.py

```diff
@@ -1,22 +1,41 @@
 # -*- coding: utf-8 -*-
 
 from flask import g, render_template, redirect, jsonify
-from datetime import datetime
-from coaster.views import jsonp, load_model
-from .. import app
+from coaster.views import jsonp, load_model, render_with
+from .. import app, funnelapp
 from ..models import Profile, ProposalSpace, Proposal
 from .space import space_data
 
 
+def index_jsonify(data):
+    spaces_data = list()
+    for space in data['spaces']:
+        space_dict = dict(space.current_access())
+        if space_dict:
+            spaces_data.append(space_dict)
+    return jsonify(spaces=spaces_data)
+
+
 @app.route('/')
+@render_with({'text/html': 'index.html.jinja2', 'application/json': index_jsonify})
 def index():
     g.profile = None
     g.permissions = []
     spaces = ProposalSpace.fetch_sorted().filter(ProposalSpace.profile != None).all()
-    return render_template('index.html.jinja2', spaces=spaces)
+    return dict(spaces=spaces)
+
+
+@funnelapp.route('/', endpoint='index')
+@render_with({'text/html': 'funnelindex.html.jinja2', 'application/json': index_jsonify})
+def funnelindex():
+    g.profile = None
+    g.permissions = []
```

Comment:
`g.permissions` is used in a lot of places on this repo. I'm not completely sure about the effects of change to `current_auth.permissions`. can we do this part of another PR? or maybe the next PR.

Reply:
`current_auth.permissions` is guaranteed to be an `InspectableSet`. `g.permissions` only promises to support the `in` operator. It could be a list, tuple or set. This difference is crucial for the transition.